src/TrajectoryPlanner/src/TrajectoryGenerator.cpp

```diff
@@ -496,6 +497,18 @@ bool TrajectoryGenerator::isTrajectoryAsked()
     return m_generatorState == GeneratorState::Called;
 }
 
+bool TrajectoryGenerator::generateTrajectoriesFromFootprintsStepAdjustment(std::shared_ptr<FootPrint> left, std::shared_ptr<FootPrint> right,
+                                                             const double &initTime,DCMInitialState initialState)
+{
+    if (!m_dcmGenerator->setDCMInitialState(initialState))
+    {
+        yError() << "[TrajectoryGenerator_Thread] Failed to set the initial state.";
+        return  false;
+    }
+
+    return m_trajectoryGenerator.generateFromFootPrints(left, right, initTime, m_dT);
```

Comment:
Please, deal also with the ``GeneratorState``. You should set it to ``GeneratorState::Returned`` if ``generateFromFootPrints`` returns ``true``. In fact, this is like requesting a new trajectory. If that method fails, you should set it to ``GeneratorState::Configured``.

Reply:
Done!